src/plugins/intel_gpu/include/intel_gpu/primitives/eye.hpp

```diff
@@ -0,0 +1,37 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include "primitive.hpp"
+
+namespace cldnn {
+/// @addtogroup cpp_api C++ API
+/// @{
+/// @addtogroup cpp_topology Network Topology
+/// @{
+/// @addtogroup cpp_primitives Primitives
+/// @{
+
+/// @brief
+/// @details
+struct eye : public primitive_base<eye> {
+    CLDNN_DECLARE_PRIMITIVE(eye)
+
+    /// @brief Constructs eye primitive.
+    /// @param id This primitive id.
+    /// @param inputs List of primitive ids.
+    eye(const primitive_id& id,
+        const std::vector<primitive_id>& inputs,
+        const tensor output_shape,
```

Comment:
I think const ref is still better here, isn't it?

Reply:
done